Sources/ComposableArchitecture/Store.swift

```diff
@@ -742,7 +559,10 @@ public final class Store<State, Action> {
   ///   .sink { ... }
   /// ```
   public var publisher: StorePublisher<State> {
-    StorePublisher(store: self, upstream: self.stateSubject)
+    StorePublisher(
+      store: self,
+      upstream: self.rootStore.didSet.map { self.currentState }
```

Comment:
Not sure if it matters, but maybe this should be:

```suggestion
      upstream: self.rootStore.didSet.compactMap { [weak self] in self?.currentState }
```